server/engine/src/main/java/org/eclipse/lsp/cobol/lsp/analysis/AsyncAnalysisService.java

```diff
@@ -46,7 +46,7 @@ public class AsyncAnalysisService implements AnalysisStateNotifier {
   private final SubroutineService subroutineService;
   private final Communications communications;
 
-  private final Map<String, CompletableFuture<CobolDocumentModel>> analysisResults = Collections.synchronizedMap(new HashMap<>());
+  private final Map<String, FutureTask<CobolDocumentModel>> analysisResults = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Use FutureTask instead of CompletableFuture as per documentation.

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html

> Since (unlike [FutureTask](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/FutureTask.html)) this class has no direct control over the computation that causes it to be completed, cancellation is treated as just another form of exceptional completion. Method [cancel](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#cancel-boolean-) has the same effect as completeExceptionally(new CancellationException()). Method [isCompletedExceptionally()](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#isCompletedExceptionally--) can be used to determine if a CompletableFuture completed in any exceptional fashion.

Reply:
Just to make sure, the `cancel` method not only requests the cancellation, but also waits for the cancellation to be completed?
Should the code also handle the possible failure of `cancel` (it may return false apparently)?